### PR TITLE
970 document derive var age years output@devel Closes #970

### LINKED ISSUE #970: Documentation:  derive_var_age_years output type of number
### Please select a category the issue is focused on?

Function Documentation

### Let us know where something needs a refresh or put your idea here!

derive_var_age_years  documentaion does not specify what is the format of the output variable (comparing to AAGE where it is stated it will be integer). 

In this function the result is a number with decimal places - in case if it is intended then in my opinion it should be specified why, I was expecting an integer as an output, not really sure why it is not (in the code we round tolower only the unit_var but not the new_var )

Link to: 
See also -> derive_vars_duration() 
should be added in the function documentation, the  derive_var_age_years is actually doing the same but the output is not integer

### PR DESCRIPTION
**Pull request by sophie-gem**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/derive_vars_aage.R [hunk 1]
```diff
@@ -86,30 +86,47 @@ derive_vars_aage <- function(dataset,
 
 #' Derive Age in Years
 #'
-#' @details This function is used to convert age variables into years.
-#' These can then be used to create age groups.
+#' Converts the given age variable (`age_var`) to the unit 'years' from the current
+#' units given in the `{age_var}U` variable or `age_unit` argument and stores
```
Comment (bms63) [inline R/derive_vars_aage.R:90]: What is the `{age_var}U`?

File: R/derive_vars_aage.R [hunk 2]
```diff
@@ -86,30 +86,47 @@ derive_vars_aage <- function(dataset,
...
+#'   in `dataset`. This gives the unit of the `age_var` variable and is used to convert
+#'   AGE to 'years' so that grouping can occur.
 #'
 #'   Default: NULL
 #'
 #'   Permitted Values: 'years', 'months', 'weeks', 'days', 'hours', 'minutes', 'seconds'
 #'
-#' @param new_var New AGE variable to be created in years.
+#' @param new_var New age variable to be created in years. The output values here are
+#'   decimal and NOT integer.
```
Comment (bms63) [inline R/derive_vars_aage.R:112]:
```suggestion
#' @param new_var New age variable to be created in years. The returned values are
#'   decimals and NOT integers.
```

File: R/derive_vars_aage.R (hunk 1 above)
Comment (sophie-gem) [reply R/derive_vars_aage.R:90]: Sorry, I didn't think this was particularly clear when I wrote it but wasn't quite sure how else to word it...

Basically, if `age_var = AGE` then `{age_var}U = AGEU`. (I was following somewhat of the `glue` notation here but am happy to change if you have an idea as to what would be clearer?)

File: R/derive_vars_aage.R (hunk 2 above)
Comment (sophie-gem) [reply R/derive_vars_aage.R:112]: Will update 👍

**Comment by sophie-gem** [comment]
> Hi @sophie-gem. apologies for just getting to this. I think this is a newsworthy update. Can you put in a short blurb in `News.md` for us.?
> 
> Also instead of decimal should we say float ?

Hi @bms63, no problems at all - I appreciate you are very busy. 

Yes of course I can update `News.md` - I have to admit I didn't as I thought it was just updating documentation and not anything from within the function...

Is float the 'technical' term for decimal in R? I know it definitely is in Python (my terminology between SAS, R, and Python gets mixed up sometimes!)

File: R/derive_vars_aage.R (hunk 1 above)
Comment (bms63) [reply R/derive_vars_aage.R:90]: ah I see.  Maybe something more explicit as some might not be familiar with that syntax.  Perhaps a `age_var+U`

**Comment by bms63** [comment]
> > Hi @sophie-gem. apologies for just getting to this. I think this is a newsworthy update. Can you put in a short blurb in `News.md` for us.?
> > Also instead of decimal should we say float ?
> 
> Hi @bms63, no problems at all - I appreciate you are very busy.
> 
> Yes of course I can update `News.md` - I have to admit I didn't as I thought it was just updating documentation and not anything from within the function...
> 

I like pointing out to users around changes to our documentation for our functions.  Nice to see in the changelog that we take documentation seriously ! :)   

> Is float the 'technical' term for decimal in R? I know it definitely is in Python (my terminology between SAS, R, and Python gets mixed up sometimes!)

Not sure on technical term...but float makes more sense to me.

**Comment by bms63** [comment]
Hey @sophie-gem are you able to finish this PR today or early tomorrow?  We are trying to release on Friday.

File: R/derive_vars_aage.R (hunk 1 above)
Comment (sophie-gem) [reply R/derive_vars_aage.R:90]: Makes sense - will update to that, thanks

**Comment by sophie-gem** [comment]
> I like pointing out to users around changes to our documentation for our functions. Nice to see in the changelog that we take documentation seriously ! :)

Very true!

**Comment by sophie-gem** [comment]
> > > Is float the 'technical' term for decimal in R? I know it definitely is in Python (my terminology between SAS, R, and Python gets mixed up sometimes!)
> > 
> > 
> > Not sure on technical term...but float makes more sense to me.
> 
> @sophie-gem , @bms63 , `typeof(1.5)` returns `"double"`. So maybe we should use "double".

Right ok - I shall replace with "double" then :)

**Comment by sophie-gem** [comment]
> Hey @sophie-gem are you able to finish this PR today or early tomorrow? We are trying to release on Friday.

Apologies @bms63 , will get this done this evening.

**Comment by sophie-gem** [comment]
@bms63 / @bundfussr  - what do you think about the following?
```
#' @details This function is used to convert an age variable into the unit 'years'
#'   which can then be used to create age groups. The resulting column contains the
#'   equivalent years in decimal. Note, underlying computations assume an equal number
#'   of days in each year (365.25). 
```

Should the use of 'decimal' here also be changed to 'The resulting column contains the equivalent years as a double.'?

**Comment by bms63** [comment]
> @bms63 / @bundfussr - what do you think about the following?
> 
> ```
> #' @details This function is used to convert an age variable into the unit 'years'
> #'   which can then be used to create age groups. The resulting column contains the
> #'   equivalent years in decimal. Note, underlying computations assume an equal number
> #'   of days in each year (365.25). 
> ```
> 
> Should the use of 'decimal' here also be changed to 'The resulting column contains the equivalent years as a double.'?

sounds good - equivalent years as a double

**Comment by bms63** [comment]
You need to do a `devtools::document()` as well to fix that last man check fail

**Comment by zdz2101** [comment]
@bms63 looks wrapped up except for the check related to https://github.com/pharmaverse/admiral/issues/1780

**Pull request merged**